projects/angular/src/data/datagrid/datagrid-column.ts

```diff
@@ -307,6 +314,10 @@ export class ClrDatagridColumn<T = any>
     return this.wrappedInjector.get(WrappedColumn, this.vcr).columnView;
   }
 
+  get datagridElementRef() {
+    return this.elementRef?.nativeElement?.closest('clr-datagrid');
+  }
```

Comment:
1. This returns the element not an `ElementRef`, so please drop `Ref` from the property name.
2. This will run the query every time the property is called. To prevent it from querying every time a filter is changed, you can cache the value in the `listenForFilterChanges`. And since it's only used there, I would just inline the query.
3. Further, it would still run this query for every column. That is unnecessary. I would instead inject the `ClrDatagrid` component, make the `ClrDatagrid#el` property `public`, and use that instead.

Reply:
I added a reference in the columns service as injecting it creates a lot of dependency issues on tests and other places.  Maybe it's better to have it in separate datagrid service, tell me what you think.

Reply:
Yeah, circular dependencies are annoying.